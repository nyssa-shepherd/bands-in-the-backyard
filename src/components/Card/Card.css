.card {
  background-color: white;
  box-shadow: 5px 5px 5px rgba(135, 135, 135, 0.586);
  height: 440px;
  margin: 19px 12px;
  width: 300px;
}

.artist-card {
  align-items: center;
  border: none;
  box-shadow: none;
  border-bottom: 2px solid rgba(164, 164, 164, 0.291);
  display: flex;
  flex-direction: row;
  height: 280px; 
  margin-left: 19%;
  padding-left: 4%;
  padding-bottom: 30px;
  width: 50%;
}

.artist-card img {
  height: 270px;
  padding-right: 35px;
}

.artist-card .artist-name {
  font-size: 30px;
  padding-right: 0;
}

.artist-card p {
  font-size: 20px;
}

.artist-card a {
  font-size: 18px;
}

.card-text {
  padding: 20px;
}

.artist-name {
  color: #1e4547;
  font-size: 18px;
  padding-bottom: 10px;
}

.location-name {
  color: #1a3c3eb3;
  font-size: 15px;
  padding-bottom: 0;
}

img {
  height: 150px;
  width: 300px;
}

p {
  color:#1a3c3eb3;
}

a {
  color: #1a3c3eb3;
  font-size: 12px;
  text-align: justify;
}

a:hover {
  color: #4bbfc3;
}

