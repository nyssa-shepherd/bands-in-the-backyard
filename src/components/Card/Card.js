import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import './Card.css';

const Card = ({name, city, state, date, time, venue, type, ticketUrl, image}) => {
  let cardtype;
  type === 'artist-card' ? cardtype = 'artist-card' : cardtype = 'event-card';
  
  return (
    <div className={`card ${cardtype}`}>
      <img src={image} />
      <div className='card-text'>
        <h1 className='artist-name'>{name}</h1>
        <h2 className='location-name'>{city}, {state}</h2>
        <p>{date} at {time}</p>
        <p>Venue: {venue}</p>
        <p>Get tickets: <a href={ticketUrl}>{ticketUrl}</a></p>
      </div>
    </div>
  );
};

export default withRouter(Card);

Card.propTypes = {
  name: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  state: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  venue: PropTypes.string.isRequired,
  ticketUrl: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};