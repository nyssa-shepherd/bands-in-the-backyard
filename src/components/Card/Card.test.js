import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card.js';

describe('Card', () => {

  it('snapshot test', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper).toMatchSnapshot();
  });

});