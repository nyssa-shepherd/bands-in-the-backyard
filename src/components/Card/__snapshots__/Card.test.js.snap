// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          src={undefined}
/>,
        <div
          className="card-text"
>
          <h1
                    className="artist-name"
          />
          <h2
                    className="location-name"
          >
                    , 
          </h2>
          <p>
                     at 
          </p>
          <p />
          <a
                    href="{ticketurl}"
          />
</div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="artist-name"
/>,
            <h2
              className="location-name"
>
              , 
</h2>,
            <p>
               at 
</p>,
            <p />,
            <a
              href="{ticketurl}"
/>,
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "artist-name",
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                ", ",
                undefined,
              ],
              "className": "location-name",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              ", ",
              undefined,
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                " at ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              " at ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "href": "{ticketurl}",
            },
            "ref": null,
            "rendered": null,
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            src={undefined}
/>,
          <div
            className="card-text"
>
            <h1
                        className="artist-name"
            />
            <h2
                        className="location-name"
            >
                        , 
            </h2>
            <p>
                         at 
            </p>
            <p />
            <a
                        href="{ticketurl}"
            />
</div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="artist-name"
/>,
              <h2
                className="location-name"
>
                , 
</h2>,
              <p>
                 at 
</p>,
              <p />,
              <a
                href="{ticketurl}"
/>,
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "artist-name",
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  ", ",
                  undefined,
                ],
                "className": "location-name",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                ", ",
                undefined,
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " at ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " at ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "href": "{ticketurl}",
              },
              "ref": null,
              "rendered": null,
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
